src/main/java/hanshyn/onlinebookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+package hanshyn.onlinebookstore.service.order;
+
+import hanshyn.onlinebookstore.dto.order.OrderItemResponseDto;
+import hanshyn.onlinebookstore.dto.order.OrderPlaceRequestDto;
+import hanshyn.onlinebookstore.dto.order.OrderResponseDto;
+import hanshyn.onlinebookstore.dto.order.OrderUpdateStatusRequestDto;
+import hanshyn.onlinebookstore.exception.EntityNotFoundException;
+import hanshyn.onlinebookstore.mapper.OrderItemMapper;
+import hanshyn.onlinebookstore.mapper.OrderMapper;
+import hanshyn.onlinebookstore.model.CartItem;
+import hanshyn.onlinebookstore.model.Order;
+import hanshyn.onlinebookstore.model.OrderItem;
+import hanshyn.onlinebookstore.model.ShoppingCart;
+import hanshyn.onlinebookstore.model.StatusOrder;
+import hanshyn.onlinebookstore.model.User;
+import hanshyn.onlinebookstore.repository.order.OrderItemRepository;
+import hanshyn.onlinebookstore.repository.order.OrderRepository;
+import hanshyn.onlinebookstore.repository.order.StatusOrderRepository;
+import hanshyn.onlinebookstore.repository.shopping.CartItemRepository;
+import hanshyn.onlinebookstore.repository.shopping.ShoppingCartRepository;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final StatusOrderRepository statusOrderRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Transactional
+    @Override
+    public OrderResponseDto createOrder(OrderPlaceRequestDto orderPlaceRequestDto) {
+        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new EntityNotFoundException("Can't found user by usr_id: " + user.getId())
+        );
+
+        StatusOrder statusOrder = statusOrderRepository.findByStatus(StatusOrder.Status.PENDING);
```

Comment:
Please check if you can use static import to avoid mentioning path
```suggestion
        StatusOrder statusOrder = statusOrderRepository.findByStatus(PENDING);
```